### DISCUSSION TITLE
CPACS 2 ribs locations

### DISCUSSION THREAD

**Discussion by GodertdeZoetenGKNFokker** [DLR-SL/CPACS#743] [Q&A] [answered]
The CPACS 2.3.1 documentation explicitly states that the eta_start and eta_end coordinates of the wingRibsPositioningType are relative to the wing eta coordinate. However, when I generate the ribs this way and compare them with the geometry generated by TIGL, I get different results. When I use an eta coordinate relative to the ribReference line, I do seem to get the same results as TIGL.

So I was wondering: is this a mistake in the CPACS 2.3.1 documentation, does TIGLviewer interpret this incorrectly or am I misunderstanding the eta coordinate?

I have attached the relevant CPACS file. 
[AGILE_DC2.zip](https://github.com/DLR-SL/CPACS/files/5438070/AGILE_DC2.zip)

**Comment by MarAlder** [comment, marked as answer]
Please excuse the late answer to this question!

From CPACS 2 to 3 there has been a significant change in the definition of coordinates in the component segments. I assume that you are aware of it, but just for comparison the [old](https://www.cpacs.de/documentation/CPACS_2_3_1_Docs/html/6a814e1f-0222-c6b8-faca-6a5efae44954.htm) and the [new](https://www.cpacs.de/documentation/CPACS_3_2_0_Docs/html/6a814e1f-0222-c6b8-faca-6a5efae44954.htm) definition.

With regard to the `wingRibsPositioningType`, in CPACS 3.2 the documentation has been updated. There it is said that the reference line is used for the rib spacing (see [here](https://www.cpacs.de/documentation/CPACS_3_2_0_Docs/html/893c8088-24d2-8047-ed69-1d4cf9d883ff.htm)): 
> The ribReference is the reference line for the computation of the rib set spacing. It can either be a sparUID or "trailingEdge" or "leadingEdge"

and `etaStart`/`etaEnd`:
> EtaStart defines the innerBorder location of the beginning of the rib set on the ribReference line. Please note, the eta is relative to wing eta coordinate! 

From this I would say that CPACS 3 now fits to your original interpretation and that the documentation is complete. Is this sufficient for your tasks or do you depend on CPACS 2 for your current activities?

Unfortunately I lack some practice with CPACS 2 (before my time as CPACS coordinator) and how TiGL interpreted it back then, so I try to reconstruct from comparision between CPACS 2 and 3. I will try to get further feedback from CPACS 2 users and come back to you.
